build/RepositoryBuild.targets

```diff
@@ -0,0 +1,69 @@
+<Project>
+  <Import Project="$(_BuildGraphFile)" />
+
+  <Target Name="BuildRepositories">
+    <ItemGroup>
+      <BatchedRepository Include="$(MSBuildProjectFullPath)">
+        <AdditionalProperties>
+          RepositoryToBuild=%(RepositoryToBuildInOrder.Identity);
+          BuildRepositoryRoot=$(_CloneRepositoryRoot)%(RepositoryToBuildInOrder.Identity)\
+        </AdditionalProperties>
+        <BuildGroup>%(RepositoryToBuildInOrder.Order)</BuildGroup>
+      </BatchedRepository>
+    </ItemGroup>
+
+    <PropertyGroup>
+      <BatchBuilds Condition="$(BuildInParallel) AND '$(CompileOnly)'=='true'">true</BatchBuilds>
+      <BatchBuilds Condition="'$(BatchBuilds)'==''">false</BatchBuilds>
+    </PropertyGroup>
+
+    <MSBuild
+      Projects="@(BatchedRepository)"
+      BuildInParallel="$(BatchBuilds)"
+      Targets="_BuildRepository"
+      Properties="BuildGroup=%(BatchedRepository.BuildGroup)" />
+  </Target>
+
+  <Target Name="_BuildRepository" DependsOnTargets="_PinVersions">
+    <PropertyGroup>
+      <BuildScriptToExecute Condition="'$(OS)'!='Windows_NT'">build.sh</BuildScriptToExecute>
```

Comment:
Instead of MSBuild => shell script => MSBuild, have you considered going straight to using MSBuild? KoreBuild is designed to key all locations off the `RepositoryRoot` property.

```xml
<MSBuild Projects="$(MSBuildProjectFullPath)" Properties="RepositoryRoot=$(BuildRepositoryRoot)" />
```

This avoids the pre-flight restore N+1 times for building N repositories, takes advantage of using the MSBuild already loaded in memory, and potentially avoids some thrashing by letting MSBuild orchestrate the entire build top to bottom.